planner/core/plan.go

```diff
@@ -283,13 +299,71 @@ func (p *basePhysicalPlan) ExtractCorrelatedCols() []*expression.CorrelatedColum
 	return nil
 }
 
+func (p *baseLogicalPlan) GetBakTimeStamp() time.Time {
+	return time.Now()
+}
+
+func (p *baseLogicalPlan) enableBak() {
```

Comment:
Can we put the `needBak` into session variable?
Since once we set it, all the sub tree is affected.

Reply:
no, I fact that's why I designed `needBak`: to reduce memory cost and avoid speed regression. During a query with `nth_plan` hint, not all nodes need to be rolled back. It serves as a sign to help indicates which node needs to be recorded with log(Bak).